telethon/client/updates.py

```diff
@@ -424,7 +425,9 @@ async def _dispatch_update(self: 'TelegramClient', update, others, channel_id, p
             if not builder.resolved:
                 await builder.resolve(self)
 
-            if not builder.filter(event):
+            filter = builder.filter(event)
+            filter = (await filter) if inspect.isawaitable(filter) else filter
```

Comment:
```suggestion
            if inspect.isawaitable(filter):
                filter = await filter
```

Reply:
You prefer full if?
Can be done :D

Reply:
It's more consistent with other places and avoids reassigning unless necessary.